### PR TITLE
test: refactor icon snaps

### PR DESCRIPTION
**Pull request by craigbilner**
Creating snapshots for icons by mocking the `Svg` libs is tricky so while they are still integration tests that will break should anything change in the svg deps, the tests are now focused on specific functionality.

Have also added a new feature to `jest-serializer` to replace a prop you'd rather be presented as something else. In this case, there are numerous svg props that are completely unreadable but you still want to know if you "break" them in any way. By replacing these props with hashes we can make them both human readable and pass `jest-lint` as well as fail on changes.

* test: refactor snaps, split and compress
* feat: add `replaceProp`
* chore: temporarily ruining some snapshots which were benefiting from a side effect in `minimalise`

### CODE REVIEW

File: packages/author-profile/__tests__/web/__snapshots__/author-profile.web.test.js.snap [hunk 1]
```diff
@@ -78,9 +78,11 @@ exports[`1. Render an author profile page 1`] = `
           <div>
             <svg
               height={15}
+              viewBox="-354.2 -279.4 750 600"
               width={15}
             >
               <path
+                d="M391.3-210.7c-28.9,10.9-57.8,21.7-86.7,25.3c32.5-18.1,54.2-50.6,68.7-83.1c-28.9,18.1-61.5,28.9-97.6,36.1 c-28.9-28.9-68.7-47-112-47c-83.1,0-151.8,68.7-151.8,151.8c0,10.8,0,25.3,3.6,36.1c-126.5-7.3-238.6-68.7-314.5-159 c-14.5,21.7-21.7,50.6-21.7,75.9c0,54.2,25.3,101.2,68.7,126.5c-25.3,0-47-7.2-68.7-18.1c0,0,0,0,0,3.6 c0,72.3,54.2,137.4,122.9,148.2c-14.5,3.6-25.3,3.6-39.7,3.6c-10.9,0-18.1,0-28.9-3.6c18.1,61.5,75.9,104.8,141,104.8 c-50.6,39.8-119.3,65.1-188,65.1c-10.8,0-25.3,0-36.1-3.6c68.7,43.3,148.2,68.7,234.9,68.7C167.2,320.6,319,89.3,319-113.1 c0-7.2,0-14.4,0-18.1C344.3-152.9,369.6-181.8,391.3-210.7z"
```
Comment (colincclark) [inline packages/author-profile/__tests__/web/__snapshots__/author-profile.web.test.js.snap]: jest-lint?

File: packages/icons/__tests__/shared-colour.base.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import "react-native";
```
Comment (colincclark) [inline packages/icons/__tests__/shared-colour.base.js]: remove

File: packages/icons/__tests__/shared-dim.base.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import "react-native";
```
Comment (colincclark) [inline packages/icons/__tests__/shared-dim.base.js]: another

File: packages/jest-serializer/README.md [hunk 4]
```diff
@@ -195,10 +195,21 @@ Refer to the code for the latest removed values.
...
-cases you may find it desirable to simply switch out bits of your tree for
-brevity and focus independent of what `jest.mock` provides or because you don't
-want to globally mock the dependency.
+A serializer that may be best suited to `jest.mock` but in some cases you may
+find it desirable to simply switch out bits of your tree for brevity and focus
+independent of what `jest.mock` provides or because you don't want to globally
+mock the dependency. `jest-mock` also doesn't play well with web or named member
+imports.
+
+###replaceProp
```
Comment (colincclark) [inline packages/jest-serializer/README.md]: needs space after ### for formatting

File: packages/jest-serializer/__tests__/ios/__snapshots__/minimal.test.js.snap [hunk 5]
```diff
@@ -42,11 +40,13 @@ exports[`The minimal serializer should for native remove undefined props 1`] = `
...
-  <Text
-    prop={<Path />}
-  >
-    Nested
-  </Text>
-</View>
+<Foo
+  prop={
+    <Text
+      dir="dir"
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/ios/__snapshots__/minimal.test.js.snap]: can we remove this?

File: packages/jest-serializer/__tests__/ios/replace-prop.test.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { addSerializers, enzymeRootSerializer, replaceProp } from "../../src";
+import shared from "../shared.replace-prop";
+
+describe("The replace prop serializer should", () => {
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/ios/replace-prop.test.js]: lower case all the things

File: packages/author-profile/__tests__/web/__snapshots__/author-profile.web.test.js.snap (hunk 1 above)
Comment (craigbilner) [reply packages/author-profile/__tests__/web/__snapshots__/author-profile.web.test.js.snap]: Referenced in the desc - need to break these before fixing them

**Pull request merged**